service/worker/scheduler/workflow.go

```diff
@@ -1216,40 +1217,13 @@ func (s *scheduler) processBuffer() bool {
 	}
 
 	isRunning := len(s.Info.RunningWorkflows) > 0
-	tryAgain := false
 
 	action := ProcessBuffer(s.State.BufferedStarts, isRunning, s.resolveOverlapPolicy)
 
 	s.State.BufferedStarts = action.NewBuffer
 	s.Info.OverlapSkipped += action.OverlapSkipped
 
-	// Try starting whatever we're supposed to start now
-	allStarts := action.OverlappingStarts
-	if action.NonOverlappingStart != nil {
-		allStarts = append(allStarts, action.NonOverlappingStart)
-	}
-	for _, start := range allStarts {
-		if !s.canTakeScheduledAction(start.Manual, true) {
-			// try again to drain the buffer if paused or out of actions
-			tryAgain = true
-			continue
-		}
-		result, err := s.startWorkflow(start, req)
-		metricsWithTag := s.metrics.WithTags(map[string]string{
-			metrics.ScheduleActionTypeTag: metrics.ScheduleActionStartWorkflow})
-		if err != nil {
-			s.logger.Error("Failed to start workflow", "error", err)
-			metricsWithTag.Counter(metrics.ScheduleActionErrors.Name()).Inc(1)
-			// TODO: we could put this back in the buffer and retry (after a delay) up until
-			// the catchup window. of course, it's unlikely that this workflow would be making
-			// progress while we're unable to start a new one, so maybe it's not that valuable.
-			tryAgain = true
-			continue
-		}
-		metricsWithTag.Counter(metrics.ScheduleActionSuccess.Name()).Inc(1)
-		nonOverlapping := start == action.NonOverlappingStart
-		s.recordAction(result, nonOverlapping)
-	}
+	tryAgain := s.startScheduledActions(action)
```

Comment:
splitting big hard-to-test function into smaller pieces that can be tested separately (this is what I did - extract loop into separate function) is usually considered a good thing to do. This was the first thing I did

Reply:
I agree it is _usually_ a good thing. But changing workflow code is an unusual situation: the code has to do exactly the same things in the exact same order as before, otherwise you get determinism errors. I agree the new code looks correct, but it's hard for me to verify that it behaves 100% identically, and if there's any doubt at all I'd rather just not do it.

Also in this case the benefit is pretty limited since it works fine (we're not fixing a bug here) and we're rewriting this code soon.

Reply:
ok, I don't want to keep this blocked. I will revert this. This also mean I will remove test.